packages/samples/headless-react/src/components/facet/facet.class.tsx

```diff
@@ -39,24 +39,57 @@ export class Facet extends Component<FacetProps> {
     this.setState(this.controller.state);
   }
 
+  private onInput(text: string) {
+    this.controller.facetSearch.updateText(text);
+    this.controller.facetSearch.search();
+  }
+
   render() {
     if (!this.state.values.length) {
       return <div>No facet values</div>;
     }
 
     return (
       <ul>
-        {this.state.values.map((value) => (
-          <li key={value.value}>
-            <input
-              type="checkbox"
-              checked={this.controller.isValueSelected(value)}
-              onChange={() => this.controller.toggleSelect(value)}
-              disabled={this.state.isLoading}
-            />
-            {value.value} ({value.numberOfResults} results)
-          </li>
-        ))}
+        <li>
```

Comment:
I think having multiple render functions, or even using composing with different classes, could be a bit more clear.

Reply:
Agreed, just well named function would help the example readability.

```
renderInput()
renderList()
renderSearchValue()
etc.
```

Reply:
I feel this sample has become too complex, and I think multiple render functions make the sample harder to read on a doc site. Unlike an IDE where it is possible to ctrl-clikc, a reader has to scroll up and down, or control-F to follow the code.

With a single render function, it is possible to see everything in one glance, provided it is not long. Could we perhaps split it into two files similar to the ResultList/ResultLink?